libopenage/datastructure/pairing_heap.h

```diff
@@ -176,10 +179,10 @@ class PairingHeapNode : public std::enable_shared_from_this<PairingHeapNode<T, c
 	}
 
 private:
-	std::shared_ptr<this_type> first_child;
-	std::shared_ptr<this_type> prev_sibling;
-	std::shared_ptr<this_type> next_sibling;
-	std::shared_ptr<this_type> parent; // for decrease-key and delete
+	this_type* child = nullptr;
```

Comment:
I felt it made more sense as a node can only have one child.

Reply:
In the code yes, but from a conceptual view, each node still has many children. The reason it's done this way is so that the pairing heap can be represented as a binary tree. You can read more about the concept here: https://en.wikipedia.org/wiki/Left-child_right-sibling_binary_tree

So `first_child` (or `left_child`) is more accurate.